---
title: "ProEnd Pipeline 1.0"
output: github_document
bibliography: References.bib
link-citations: true
csl: https://raw.githubusercontent.com/citation-style-language/styles/master/biomed-central.csl
---
```{r, , include=FALSE}
#options(scipen=999) #non scientific ; options(scipen=0) # scientific notation
#BiocManager::install(c("biomaRt","org.At.tair.db","enrichplot","GOSemSim","clusterProfiler"))
library("biomaRt")
library(org.At.tair.db)
library(clusterProfiler)
library(GOSemSim)
library(enrichplot)
library(ggplot2)
```

# ProEnd Scripts

This project contains two Bash scripts designed to handle and analyze multiple protein sequences. The scripts streamline the extraction and identification of specific C-terminal protein HbYX-motifs from a given FASTA file. The third script shows how to download the proteomes from UniProt.

If you have just one protein sequence or your file is already one protein sequence per line go to script 2
## Scripts

### 1. HbYX-First.bash

This script formats a multi-entry FASTA file into a single line per entry format, preparing it for further analysis. Use this script first if you have multiple sequences, a proteome, or alignments.

#### Usage

Ensure you have the necessary FASTA file in the same directory or specify the path to the file. Execute the script by running:

`./HbYX-First.bash`

or 

`bash HbYX-First.bash `

This will output a file named `multifasta.ol.fa`, containing all the sequences from the original FASTA file, formatted for further processing.

### 2. HbYX-Second-Script.bash

This script searches for a specific HbYX-motif at the C-terminus of the protein sequences in the `multifasta.ol.fa` file created by the first script.

#### Usage

Run the script using:

`./HbYX-Second-Script.bash`

or 

`bash HbYX-Second-Script.bash`

It will produce a file named `2aa.txt`, containing the motifs found along with the corresponding header from the FASTA file, if available.

### Testing ProEND with *Arabidopsis proteome*

Download (Reviewed Swiss-Pro) Arabidopsis [proteome from Uniprot](https://www.uniprot.org/uniprotkb?query=arabidopsis&facets=reviewed%3Atrue)

#### 1. Fasta linearization

```{bash, eval=TRUE}
zcat Arabidopsis_uniprot_proteome/uniprotkb_arabidopsis_AND_reviewed_true_2024_10_08.fasta.gz | head -n 2
```

```{bash, eval=TRUE}
bash HbYX-First.awk Arabidopsis_uniprot_proteome/uniprotkb_arabidopsis_AND_reviewed_true_2024_10_08.fasta.gz Arabidopsis_uniprot_proteome/arabidopsis_uniprot_proteome.ol.fa
``` 

```{bash, eval=TRUE, include=FALSE}
head -n 2 Arabidopsis_uniprot_proteome/arabidopsis_uniprot_proteome.ol.fa
#Removing non Arabidopsis sequences 
grep -A 1 --no-group-separator "OS=Arabidopsis thaliana"  Arabidopsis_uniprot_proteome/arabidopsis_uniprot_proteome.ol.fa >  Arabidopsis_uniprot_proteome/arabidopsis_uniprot_proteome.ol_clean.fa
```

#### 2. HbYX motif prediction 
```{bash, eval=TRUE}
bash HbYX-Second-Script.awk  Arabidopsis_uniprot_proteome/arabidopsis_uniprot_proteome.ol_clean.fa  Arabidopsis_uniprot_proteome/arabidopsis_HbyX_proteome.txt
```

Total number of HbYX motif candidates: 
```{bash, eval=TRUE}
grep -c ">" Arabidopsis_uniprot_proteome/arabidopsis_HbyX_proteome.txt 
```

#### 3. Exploring conserved HbYX proteasome regulatory protein

```{bash, eval=TRUE}
grep --no-group-separator -A 1 -i "proteasome"  Arabidopsis_uniprot_proteome/arabidopsis_HbyX_proteome.txt 
```

#### 4. Expanding to new regulatory candidates

Mapping TAIR Arabidopsis IDs using biomaRt

```{bash, eval=FALSE, include=FALSE}
grep ">"  Arabidopsis_uniprot_proteome/arabidopsis_HbyX_proteome.txt | awk -F'|' '{print $2}' > Arabidopsis_uniprot_proteome/uniprot_ids.txt
```
Getting entrezID and running GO terms analysis 

```{r,  include=TRUE}
uniprot_ids <- read.csv("Arabidopsis_uniprot_proteome/uniprot_ids.txt", header = F)
ensembl <- useMart("plants_mart", dataset = "athaliana_eg_gene", host = "https://plants.ensembl.org")
mapping <- getBM(
  attributes = c("uniprotswissprot", "entrezgene_id"), # "tair_locus"),
  filters = "uniprotswissprot",
  values = uniprot_ids,
  mart = ensembl
)

all_uniprot_df <- as.data.frame( uniprot_ids)
colnames(all_uniprot_df ) <- c("uniprotswissprot")

uniprot2entrez <- merge(all_uniprot_df, mapping, by = "uniprotswissprot", all.x = TRUE)
print(paste("Uniprot ids:", nrow(uniprot_ids), " Entrez ids:", nrow(mapping), " Non mapped ids:",  nrow(uniprot_ids) -  nrow(mapping), sep = ""))
#Non mapped IDs
#uniprot2TAIR[is.na(uniprot2TAIR$tair_locus),]

ego <- enrichGO(gene  = uniprot2entrez[!is.na(uniprot2entrez$entrezgene_id),2], 
                OrgDb         = org.At.tair.db,
                ont           = "BP",
                pAdjustMethod = "BH",
                pvalueCutoff  = 0.01,
                qvalueCutoff  = 0.05,
                readable      = TRUE) #library(clusterProfiler)

ego_at <- attributes(ego )
Whole_table <- ego_at$result
write.csv(Whole_table, "Arabidopsis_uniprot_proteome/GO_HbYX_Arabidopsis.csv") 
d <- godata('org.At.tair.db', ont="BP") #library(GOSemSim)

ego2 <- pairwise_termsim(ego, method="Wang", semData = d) #library(enrichplot)
```
```{r,  include=FALSE}
gos_clustered <-  treeplot(ego2 )

ggsave(filename = "Arabidopsis_uniprot_proteome/ara_go_tree.svg", plot = gos_clustered,device = "svg", width=13, height=7)
row.names(Whole_table) <- NULL
```
<div style="text-align: center;">
<figure>
<img src="Arabidopsis_uniprot_proteome/ara_go_tree.svg"
style="width: 100%;height: auto">
<figcaption style="margin-top: 10px;"><strong>GO terms enrichment of Arabidopsis HbYX contaiting proteins</strong></figcaption>
</figure>
<a name="GO_terms_HbYX"></a>
</div>

5. Testing CDC48A as HbYX proteasome regulatory protein

One of the lesser-known interactors of the 26S proteasome, significantly enriched across several Gene Ontology categories, is the CDC48 gene family, as demonstrated in:
<!-- CDC48A (At3g09840) from Arabidopsis -->
```{r, eval=TRUE}
head(Whole_table[grepl("CDC48",Whole_table$geneID),c(2,3,8)])
```


## Requirements

- Unix-like environment
- AWK installed

## Installation

No installation is required. Simply clone this repository or download the scripts to your local machine.

## Data Folder HbYX_data_tables
This folder contains data results files for the ProEnd Scripts project.

## Cite 
This code can be cited currently as BioRxiv preprint @salcedo2024proend

## References 